### PR TITLE
feat(timm_regnet): add timm RegNet image-classification family

### PR DESCRIPTION
**Pull request by zhenshanx-nv**
## Background

RegNet is the last of the tensorrtx classifier baselines that reuses the
existing convolutional op set. `timm/regnety_040.ra3_in1k` cannot be built or
served today.

## Exit Criteria

- A `timm_regnet` family builds timm RegNet checkpoints from HF-hosted
  safetensors and produces logits matching timm's own implementation.
- RegNetX (no squeeze-excite) and RegNetY (with it) build from one code path.
- The family is registered across the runtime strategy matrix, validation
  workloads, benchmark suite, website data, and the E2E model registry.

Non-goals: quantized builds, tensor-parallel builds, and the RegNetZ variants,
which change the block structure.

## Implementation

The layout is recovered from the checkpoint: stages and block counts from the
`s<stage>.b<block>` keys, the convolution group count from the 3x3 weight shape,
and the squeeze-excite gate and downsample path from the keys that are present.
RegNet halves the resolution in the first block of every stage, so the stride is
a uniform rule rather than a per-model table.

Detecting the gate from the keys means RegNetX builds from the same path as
RegNetY with no extra branch.

Two shape details worth noting for reviewers:

- timm wraps each convolution and its norm in a `ConvNormAct`, so weights are
  named `conv.weight` and `bn.*` under each leaf rather than flat `convN`/`bnN`.
- The squeeze-excite projections are named `fc1`/`fc2` even though both are 1x1
  convolutions.

The gate uses a ReLU inner activation with a plain sigmoid. That is a third
combination, distinct from MobileNetV3 (hard-sigmoid, ReLU) and EfficientNet
(sigmoid, SiLU), so the family keeps its own copy rather than sharing a
parameterised helper.

No public API, ABI, or bundle format change. No new dependencies.

### Change categories

- [x] Model or runtime behavior
- [ ] Public API
- [ ] ABI
- [ ] Bundle or artifact format
- [ ] Dependencies
- [ ] Documentation only
- [ ] CI or developer tooling

### Commands and Results

```
python -m pytest tests/builder/ tests/tools/ tests/e2e_harness/ -q -n 8 \
  --dist=worksteal --import-mode=importlib -p no:cacheprovider
=> 3990 passed, 8 skipped

python -m pytest -q tests/e2e/models/timm_regnet/test_timm_regnet_family_plugin.py
=> 15 passed

cmake --build $BUILD --target trtmc_model_timm_regnet \
  test_timm_regnet_image_preprocess_seam
$BUILD/test_timm_regnet_image_preprocess_seam
=> build and link clean; test exit 0

python -m ruff check ... => All checks passed
python tools/legal_headers.py => findings=0
clang-format => clean
```

Numerical parity against timm's own implementation, which shares no code with
the builder:

| Checkpoint | Correlation | argmax | top-5 |
| --- | --- | --- | --- |
| `timm/regnety_040.ra3_in1k` | 0.99999874 | match | 5/5 |

The state dict loads into timm with no missing or unexpected keys.

### Hardware, Environment, and Revisions

- GPU: NVIDIA A100-SXM4-80GB, compute capability 8.0.
- Container: `Dockerfile.dev.x86` dev image, Ubuntu 24.04, Python 3.12.
- TensorRT 11.1.0.106, CUDA architecture `80-real`, Release build.
- Reference: timm 1.0.29 with torchvision 0.27.0+cpu on torch 2.12.0+cpu.
- Parity measured at fp32; the family also supports fp16.
- `timm/regnety_040.ra3_in1k` @ `a1875159c7b04f3b3189af804642eebba1e2c118`.

  The manifest does not pin `hf_revision`: the timm reference resolves
  `hf-hub:<id>` at `main`, so a pin disagrees with the cache the warm step
  populates and fails the offline reference run. See #1121.

### Not Run / Remaining Gaps

- No E2E harness run. The manifest is registered but was not executed here.
- Only `regnety_040` was verified numerically. RegNetX is handled by the same
  path and is unit-tested for the no-gate case, but **no RegNetX checkpoint was
  downloaded or compared**, so that path has no numerical evidence.
- The `regnetz_*` variants match the `regnet` prefix but change the block
  structure and are not supported.
- No performance numbers. The benchmark row is registered but was not run.

## Notes For Future Readers

The three squeeze-excite variants across these families differ in both the inner
activation and the gate, and none of the differences changes tensor shapes. If
you add another family with a gate, check both against timm rather than assuming
one of the existing copies applies.

### Risk level

- [x] Low
- [ ] Medium
- [ ] High

Additive family. Existing families are untouched except for shared registration
points, all widened rather than redirected, and the full CPU suite passes.

### CODE REVIEW

**Comment by zhenshanx-nv** [comment]
Superseded by #1207, which implements this family in the post-cutover layout introduced by #1188. This branch targets `python/tensorrt_model_connect/families/`, which no longer exists, so it cannot be rebased.
